crates/interledger-btp/src/server.rs

```diff
@@ -33,121 +21,133 @@ const MAX_MESSAGE_SIZE: usize = 40000;
 /// The separation is designed to enable the returned BtpOutgoingService to be passed
 /// to another service like the Router, and _then_ for the Router to be passed as the
 /// IncomingService to the BTP server.
-pub fn create_server<T, U, A>(
+pub fn btp_server<O, S, A>(
     ilp_address: Address,
-    address: SocketAddr,
-    store: U,
-    next_outgoing: T,
-) -> impl Future<Item = BtpOutgoingService<T, A>, Error = ()>
+    store: S,
+    next_outgoing: O,
+) -> (
+    BtpOutgoingService<O, A>,
+    warp::filters::BoxedFilter<(impl warp::Reply,)>,
+)
 where
-    T: OutgoingService<A> + Clone + Send + Sync + 'static,
-    U: BtpStore<Account = A> + Clone + Send + Sync + 'static,
-    A: BtpAccount + 'static,
+    O: OutgoingService<A> + Clone + Send + Sync + 'static,
+    S: BtpStore<Account = A> + Clone + Send + Sync + 'static,
+    A: BtpAccount + Account + 'static,
```

Comment:
Since BtpAccount extends Account this should be possible:
 
```suggestion
    A: BtpAccount + 'static,
```